internal/dataplane/address_finder.go

```diff
@@ -72,6 +82,28 @@ func (a *AddressFinder) GetAddresses() ([]string, error) {
 	return nil, fmt.Errorf("data-plane addresses can't be retrieved: no valid method available")
 }
 
+// GetUDPAddresses provides a list of the UDP addresses which the data-plane is
+// listening on for ingress network traffic. Addresses can either be IP
+// addresses or hostnames. If UDP settings are not configured, falls back to GetAddresses().
+func (a *AddressFinder) GetUDPAddresses() ([]string, error) {
+	a.lock.RLock()
+	defer a.lock.RUnlock()
+
+	if len(a.overrideAddressesUDP) > 0 {
+		return a.overrideAddressesUDP, nil
+	}
+
+	if len(a.overrideAddresses) > 0 && a.addressGetter == nil {
```

Comment:
What is the priority of `a.overrideAddresses` and `a.addressGetter` if they are both specified? In other words, if `len(a.overrideAddresses) > 0` and `a.addressGetter != nil`, what addresses are expected to return?

Reply:
The overrides should always take precedence, so if the overrides are there, they have priority.

Reply:
I have this should have at the very least a comment, stating that the overrides for non-UDP will also override the UDP if UDP overrides are not in place (🤯)